base_user_role_history/tests/test_res_users_role.py

```diff
@@ -0,0 +1,181 @@
+from datetime import date
+
+from odoo.addons.base.tests.common import BaseCommon
+
+
+class TestResUsersRole(BaseCommon):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.ResUsers = cls.env["res.users"]
+        cls.ResUsersRole = cls.env["res.users.role"]
+        cls.ResUsersRoleLine = cls.env["res.users.role.line"]
+        cls.BaseUserRoleLineHistory = cls.env["base.user.role.line.history"]
+
+        cls.user = cls.ResUsers.create(
+            {
+                "name": "Test user 01",
+                "login": "test1",
+            }
+        )
+
+    def test_create_adds_history(self):
+        # act
+        role_01 = self.ResUsersRole.create(
+            {"name": "Test role 01", "line_ids": [(0, 0, {"user_id": self.user.id})]}
+        )
+
+        # assert
+        self.assertTrue(role_01.name, "Test role 01")
+
+        history_lines_all = self.BaseUserRoleLineHistory.search(
+            [("user_id", "=", self.user.id)]
+        )
+        self.assertEqual(len(history_lines_all), 1)
+
+        history_line_1 = history_lines_all[0]
+        self.assertEqual(history_line_1.user_id, self.user)
+        self.assertEqual(history_line_1.performed_action, "add")
+        self.assertFalse(history_line_1.old_role_id)
+        self.assertEqual(history_line_1.new_role_id, role_01)
+        self.assertFalse(history_line_1.old_is_enabled)
+        self.assertTrue(history_line_1.new_is_enabled)
+
+    def test_write_adds_history_entry(self):
+        # arrange
+        role_01 = self.ResUsersRole.create(
+            {"name": "Test role 01", "line_ids": [(0, 0, {"user_id": self.user.id})]}
+        )
+
+        role_01_line = role_01.line_ids[0]
+
+        # act
+        role_01_line.write(
+            {
+                "date_from": "2024-05-01",
+                "date_to": "2024-05-30",
+            }
+        )
+
+        # assert
+        history_lines_all = self.BaseUserRoleLineHistory.search(
+            [("user_id", "=", self.user.id)]
+        )
+
+        self.assertEqual(len(history_lines_all), 2)
+
+        history_lines_add = history_lines_all.filtered(
+            lambda x: x.performed_action == "edit"
+        )
+        self.assertEqual(len(history_lines_add), 1)
+
+        history_line_1 = history_lines_add[0]
+        self.assertEqual(history_line_1.performed_action, "edit")
```

Comment:
Maybe we could add another assert which I found in the deleted tests. The rest I agree.
```suggestion
        self.assertEqual(history_line_1.old_role_id, role_01)
        self.assertEqual(history_line_1.new_role_id, role_01)
```